scripts/publishReleaseOnGithub.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+
+SCRIPT_LOCATION=$(dirname -- "$(readlink -f -- "${BASH_SOURCE[0]}")")
+# shellcheck source=./release_functions.sh
+source "${SCRIPT_LOCATION}/source_files/release_functions.sh"
+source "${SCRIPT_LOCATION}/source_files/helper_functions.sh"
+
+
+function create_message_body(){
+  verify_version_present_and_formatted $1 || return $?
+  version="$1"
+  release_tag="v$1"
+  is_draft=$(determine_is_draft "${RELEASE}")
+  json_adjacent_body="{
+    'tag_name' : '${release_tag}',
+    'name' : 'Release ${release_tag}',
+    'body' : 'Version ${version} release of JGiven.\nSource code may be found in Maven Central: https://search.maven.org/search?q=com.tngtech.jgiven',
+    'draft' : ${is_draft}
+  }"
+  replace_single_by_double_quotes "${json_adjacent_body}"
+  return $?
+}
+
+body=$(create_message_body "$1") || { code=$?;  echo "Failed to create message body with error ${code}"; exit ${code}; }
+action_token=$([ -n "$2" ] && echo "$2") || { echo "Failed to parse action token"; exit 1; }
+curl -X POST \
+  -H "Accept: application/vnd.github.v3+json" \
+  https://api.github.com/repos/TNG/JGiven/releases \
```

Comment:
I think this means that your user must have write access to the JGiven repository. have you ensured/checked this?

Reply:
Yes I need write access to the repo. But my JGiven token should have that